x-pack/metricbeat/module/prometheus/remote_write/config.go

```diff
@@ -21,12 +25,21 @@ var defaultConfig = config{
 	TypesPatterns: TypesPatterns{
 		CounterPatterns:   nil,
 		HistogramPatterns: nil},
+	Period: time.Second * 60,
 }
 
 func (c *config) Validate() error {
 	if c.RateCounters && !c.UseTypes {
 		return errors.New("'rate_counters' can only be enabled when `use_types` is also enabled")
 	}
-
+	duration, err := time.ParseDuration(c.Period.String())
+	{
+		if err != nil {
+			return err
+		} else if duration < 60*time.Second {
+			// by default prometheus push data with the interval 60s, in order to calculate counter rate we are setting Period to 60secs accordingly
+			c.Period = time.Second * 60
```

Comment:
Should we log a warning saying the period needs to be at least 60s?

Reply:
I would say for now that debug log [here](https://github.com/elastic/beats/pull/38553/files/6ae51f8d8b684b7e9db671deba8386a278988f5e#diff-1f5a1fd65467e6fd6069367f7043f1b33addab74d16469cbb2d86a61fab86902R47) and the comment [ here](https://github.com/elastic/beats/pull/38553/files/6ae51f8d8b684b7e9db671deba8386a278988f5e#diff-8ab9ea7fd7d76fa5a72d7eb71845defcb6c5b70f6e49dd9e9356b1fbaa40645eR101) are enough. Afterall this is an internal cache so not many users will want to change it